src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.RequestTelemetryLogger.cs

```diff
@@ -29,21 +32,28 @@ public RequestTelemetryLogger(string serverTypeName)
             {
                 _serverTypeName = serverTypeName;
                 _requestCounters = new ConcurrentDictionary<string, Counter>();
+                _histogramLogAggregator = CreateLogAggregator();
             }
 
-            public void LogSuccess(string methodName)
+            public void UpdateTelemetryData(string methodName, TimeSpan queuedDuration, TimeSpan requestDuration, Result result)
             {
-                Interlocked.Increment(ref _requestCounters.GetOrAdd(methodName, new Counter())._successfulCount);
-            }
-
-            public void LogCancel(string methodName)
-            {
-                Interlocked.Increment(ref _requestCounters.GetOrAdd(methodName, new Counter())._cancelledCount);
-            }
+                // Update the overall queue time metrics since the time in queue is not specific to the LSP method.
+                _histogramLogAggregator?.IncreaseCount(QueuedDurationKey, Convert.ToDecimal(queuedDuration.TotalMilliseconds));
 
-            public void LogFailure(string methodName)
-            {
-                Interlocked.Increment(ref _requestCounters.GetOrAdd(methodName, new Counter())._failedCount);
+                // Store the request time metrics per LSP method.
+                _histogramLogAggregator?.IncreaseCount(methodName, Convert.ToDecimal(requestDuration.TotalMilliseconds));
```

Comment:
readss odd.  you're increasing the count, but using time to add to it.  i don't really get it.

Reply:
The method name is a little odd, but it's basically increasing the count of whatever bucket the request duration falls into.  So if it's between 50-100ms, it will increase the count of the items in the 50-100 bucket.